teps/0021-results-api.md

```diff
@@ -0,0 +1,772 @@
+  publishing, Pull Request updates, etc. to be associated to their execution
+  events. It's unclear whether we would want to store this information in Runs,
+  since it is a different resource / execution.
+
+We may choose to provide a facade of the Task/Pipeline APIs in the future for
+convenience, but this would be a layer in addition to the Results API.
+
+### REST/Open API
+
+One alternative to using gRPC would be to define an Open API spec and implement
+a REST API as the surface.
+
+We are choosing to go with gRPC to take advantage of protobuf serialization,
+multi-language client/server library generation, and other performance features
+baked in to gRPC by default.
+
+For clients that require a REST API, we can configure a
+[gateway](https://github.com/grpc-ecosystem/grpc-gateway) to handle HTTP <->
+gRPC transcoding.
+
+More Reading:
+
+- https://www.redhat.com/en/blog/comparing-openapi-grpc
+- https://cloud.google.com/blog/products/api-management/understanding-grpc-openapi-and-rest-and-when-to-use-them
+
+### Making Results part of the core Pipeline controller
+
+While we expect that this will be a commonly used component for many
+installations, we do not want to make this a requirement for every installation.
+Users should be able to enable / disable this component freely to fit their
+needs.
+
+### Naming overlap with Task results
+
+There is naming overlap with
+[Task outputs](https://github.com/tektoncd/pipeline/blob/master/docs/tasks.md#emitting-results),
+called Task results. There is a risk that users may confuse Tekton Results with
+Task results.
+
+This is a particularly messy history, because the original proposal for the
+Results API came out before
+[Task results were introduced to the codebase](https://github.com/tektoncd/pipeline/pull/1921)
+and the
+[initial mention](https://github.com/tektoncd/pipeline/issues/1273#issuecomment-546494832)
+of the Task results field (named results to not conflict with PipelineResource
+outputs) was suggested the same week as the original Results API design doc.
+Both of these predated the current TEP process.
+
+At this time, we think that Results are still the best fit name for this project
+(though suggestions welcome!). Alternatives considered:
+
+- Outcome API : Likely our best candidate if we think result should reserved for
+  Task outputs. Doesn't quite capture that results maybe updated throughout the
+  lifecycle of an event (i.e. outcome feels inherently post-execution), and IMO
+  overall doesn't feel as natural as result.
+- Result History API : Similar issue to outcome in that it doesn't capture that
+  results may change over time, but might be enough of a distinction to separate
+  the project from Task output results.
+- Event API : While event is general enough to fill a similar role, this takes
+  away our usage of event as a field of the current Result resource, which then
+  leaves us with a different naming problem.
```

Comment:
"stuff that happened"

The only other name that seems appealing to me is something like "history"? "Tekton History"?